rundeckapp/grails-app/services/rundeck/services/FrameworkService.groovy

```diff
@@ -775,6 +765,30 @@ class FrameworkService implements ApplicationContextAware {
     def Description getStepPluginDescription(String type) throws MissingProviderException{
         rundeckFramework.getStepExecutionService().providerOfType(type).description
     }
+
+    /**
+     * Return step plugin of a certain type
+     * @param type
+     * @return
+     */
+    def getStepPlugin(String type) throws MissingProviderException{
+        rundeckFramework.getStepExecutionService().providerOfType(type)
+    }
+
+    /**
+     * Return dynamic properties values
+     * @param type, projectAndFrameworkValues
+     * @return
+     */
+    def Map<String, Object> getDynamicProperties(String type, Map<String, Object> projectAndFrameworkValues){
+        def plugin = getStepPlugin(type)
```

Comment:
There needs to be another call which can evaluate for "nodeStep" plugins, in addition to "step" plugins.